homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -189,7 +190,17 @@ def _calc_precipitation(rain, snow):
     @staticmethod
     def _get_condition(weather_code):
         """Get weather condition from weather data."""
-        return [k for k, v in CONDITION_CLASSES.items() if weather_code in v][0]
+        condition = [k for k, v in CONDITION_CLASSES.items() if 800 in v][0]
+
+        if condition == "sunny/clear-night":
+            hour = dt.now().hour
+
+            if hour >= 6 and hour <= 18:
```

Comment:
we can't rely on these values of day or night, for every people these values will be different

Reply:
What would be a better solution? I don't think there is another way to tell for every single timezone if it is day or night.

Reply:
hm..maybe you can find out when sun is below or above horizon.
also you can't use `dt.now()` you need to pass forecast\weather time, and calculate status according to time from API not for now

Reply:
Please see my messages further down.

Reply:
Solved by using sun.is_up() and using the corresponding unix timestamp for the forecast instead of dt.now().